### PR TITLE
Introduce efficient and parallel DelayedFields used for aggregator_v2

### PR DESCRIPTION
**Pull request by igor-aptos**
This is a feature branch, every commit added to it has been reviewed separately.

This implements [AIP-47]

### CODE REVIEW

**Review by vusirikala — APPROVED** [review]

File: aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs [hunk 1]
```diff
@@ -433,16 +458,18 @@ mod tests {
...
     }
 
     #[test]
     fn size_computation_new_group() {
-        let s = MockStateView::new(HashMap::new());
+        let s = MockStateView::new(BTreeMap::new());
         let resolver = as_resolver_with_group_size_kind(&s, GroupSizeKind::AsSum);
 
-        let group_changes = HashMap::from([(mock_tag_1(), MoveStorageOp::New(vec![2, 2].into()))]);
+        // TODO: Layout hardcoded to None. Test with layout = Some(..)
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs:470]: missed todo

File: aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs [hunk 2]
```diff
@@ -365,12 +388,12 @@ mod tests {
             resolver.resource_group_size(&key).unwrap(),
             expected_size as u64
         );
-
-        let group_changes = HashMap::from([
+        // TODO: Layout hardcoded to None. Test with layout = Some(..)
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs:391]: missed todo

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs [hunk 3]
```diff
@@ -190,7 +271,8 @@ impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
         resource_tag: &Self::ResourceTag,
         maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<Bytes>> {
-        if let Some(write_op) = self
+        // TODO: resource_group_write_set also contains a layout. What to do with it?
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:274]: RG todo

File: aptos-move/aptos-vm/src/storage_adapter/executor_view_base.rs [hunk 4]
```diff
@@ -0,0 +1,117 @@
+// Copyright © Aptos Foundation
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/storage_adapter/executor_view_base.rs:1]: I believe this is unused, will remove in the next PR

**Review by georgemitenkov — APPROVED** [review]

**Pull request merged**